File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if(x>=0):
            a=str(x)
            a=a[::-1]
            a=int(a)
            if(a >= 2 ** 31 - 1 or a <= -2 ** 31):
                return 0
            return a
        else:
            x=x*-1
            a=str(x)
            a=a[::-1]
            a=int(a)
            a=-a
            if(a >= 2 ** 31 - 1 or a <= -2 ** 31):
                return 0
            return a

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int rem = 0;

        while(x != 0) {
            int digit = x % 10;
            if((rem > Integer.MAX_VALUE / 10 || (rem == Integer.MAX_VALUE / 10 && digit > 7)) ||
                (rem < Integer.MIN_VALUE /10 || (rem == Integer.MIN_VALUE / 10 && digit < -8))) {
                    return 0;
                }
            rem *= 10;
            rem += digit;
            x /= 10;
        }

        return rem;
    }
}